app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,70 @@
+class CategoriesController < ApplicationController
+  before_action :verify_admin, except: [:index, :show, :homepage]
+  before_action :set_category, except: [:index, :homepage, :new, :create]
+
+  def index
+    @categories = Category.all.order(:name)
+  end
+
+  def show
+    @posts = @category.posts.where(post_type_id: @category.display_post_types)
+                      .includes(:post_type).list_includes.paginate(page: params[:page], per_page: 50)
+  end
+
+  def homepage
+    @category = Category.where(is_homepage: true).first
+    @posts = @category.posts.where(post_type_id: @category.display_post_types)
+                      .includes(:post_type).list_includes.paginate(page: params[:page], per_page: 50)
+    render :show
+  end
+
+  def new
+    @category = Category.new
+  end
+
+  def create
+    @category = Category.new category_params
+    if @category.save
+      if @category.is_homepage
+        Category.where.not(id: @category.id).update_all(is_homepage: false)
+      end
+      flash[:success] = 'Your category was created.'
+      redirect_to category_path(@category)
+    else
+      flash[:danger] = 'There were some errors while trying to save your category.'
+      render :new
+    end
+  end
+
+  def edit; end
+
+  def update
+    if @category.update category_params
```

Comment:
I'm getting the following error message on submission:

![error message screenshot (see below for message content)](https://user-images.githubusercontent.com/21335202/79337257-744de180-7f25-11ea-9250-2a39caa43890.png)

> can't dump `display_post_types`: was supposed to be a Array, but was a String. -- "1"